backend/src/Data/DataX.cs

```diff
@@ -63,8 +60,12 @@ DateTime createdAt
     public AppliedMethod AppliedMethod { get; private set; } = default!;
 
     public ICollection<DataApproval> Approvals { get; } = new List<DataApproval>();
-    // public ResponseApproval Approval { get; private set; }
+    public ResponseApproval? Approval { get; set; }
 
-    // TODO Exactly one resource must not have a parent and each other resource must have one from this list and the graph must be connected. In other words, the resources must form a tree.
+    // TODO Exactly one resource must not have a parent and each other resource must have one from
+    // this list and the graph must be connected. In other words, the resources must form a tree.
     public virtual ICollection<GetHttpsResource> Resources { get; } = new List<GetHttpsResource>();
+
+    public DataAccessMode DataAccess { get; set; } = DataAccessMode.UNRESTRICTED;
```

Comment:
I would prefer it if this "mode" would not be present. The mode can be inferred from whether the fields in `DataAccessRights` are `null` or not. For me it is ambiguous information if for example the mode is `UNRESTRICTED` when at the same time there are non-null fields in `DataAccessRights`. A good data model should not allow ambiguous states/information to be possible in the first place.